code/math/curve.cpp

```diff
@@ -9,6 +9,61 @@ int curve_get_by_name(const SCP_string& in_name) {
 	return find_item_with_field(Curves, &Curve::name, in_name);
 }
 
+static int curve_inline_def(){
+	//The curve needs a unique identifier. Build this with semicolon + filename + semicolon + character in file
+	SCP_string curve_name = ';' + SCP_string(Current_filename) + ';' + std::to_string(Mp - Parse_text);
+
+	int resulting_curve = static_cast<int>(Curves.size());
+	Curve& new_curve = Curves.emplace_back(curve_name);
+
+	do {
+		curve_keyframe& new_keyframe = new_curve.keyframes.emplace_back();
+		stuff_float(&new_keyframe.pos.x);
+		stuff_float(&new_keyframe.pos.y);
+		required_string(")");
+
+		bool found_interpolation = true;
+		int interpolation_mode = optional_string_one_of(4, "--", "-|", "-/", "/-");
+		switch (interpolation_mode) {
+		case 0:
+			new_keyframe.interp_func = CurveInterpFunction::Linear;
+			break;
+		case 1:
+			new_keyframe.interp_func = CurveInterpFunction::Constant;
+			break;
+		case 2:
+			new_keyframe.interp_func = CurveInterpFunction::Polynomial;
+			new_keyframe.param1 = 2.0f;
+			new_keyframe.param2 = 1.0f;
+			break;
+		case 3:
+			new_keyframe.interp_func = CurveInterpFunction::Polynomial;
+			new_keyframe.param1 = 2.0f;
+			new_keyframe.param2 = -1.0f;
+			break;
+		default:
+			new_keyframe.interp_func = CurveInterpFunction::Linear;
+			found_interpolation = false;
+			break;
+		}
+		if (!found_interpolation)
+			break;
+	} while (optional_string("("));
+
+	return resulting_curve;
+}
+
+int curve_parse() {
+	if(optional_string("(")) {
+		return curve_inline_def();
+	}
+	else {
+		SCP_string curve_name;
+		stuff_string(curve_name, F_NAME);
+		return curve_get_by_name(curve_name);
```

Comment:
I'm not a fan of removing the curve name from the `error_display` instances elsewhere, so why not add an `error_display` here?